common/djangoapps/third_party_auth/tests/specs/test_lti.py

```diff
@@ -0,0 +1,130 @@
+"""
+Integration tests for third_party_auth LTI auth providers
+"""
+import unittest
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from oauthlib.oauth1.rfc5849 import Client, SIGNATURE_TYPE_BODY
+from third_party_auth.tests import testutil
+
+FORM_ENCODED = 'application/x-www-form-urlencoded'
+
+LTI_CONSUMER_KEY = 'consumer'
+LTI_CONSUMER_SECRET = 'secret'
+LTI_TPA_LOGIN_URL = 'http://testserver/auth/login/lti/'
+LTI_TPA_COMPLETE_URL = 'http://testserver/auth/complete/lti/'
+LTI_USER_ID = 'lti_user_id'
+EDX_USER_ID = 'test_user'
+EMAIL = 'lti_user@example.com'
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class IntegrationTestLTI(testutil.TestCase):
+    """
+    Integration tests for third_party_auth LTI auth providers
+    """
+
+    def setUp(self):
+        super(IntegrationTestLTI, self).setUp()
+        self.configure_lti_provider(
+            name='Other Tool Consumer 1', enabled=True,
+            lti_consumer_key='other1',
+            lti_consumer_secret='secret1',
+            lti_max_timestamp_age=10,
+        )
+        self.configure_lti_provider(
+            name='LTI Test Tool Consumer', enabled=True,
+            lti_consumer_key=LTI_CONSUMER_KEY,
+            lti_consumer_secret=LTI_CONSUMER_SECRET,
+            lti_max_timestamp_age=10,
+        )
+        self.configure_lti_provider(
+            name='Other Tool Consumer 2', enabled=True,
+            lti_consumer_key='other2',
+            lti_consumer_secret='secret2',
+            lti_max_timestamp_age=10,
+        )
+        self.lti = Client(
+            client_key=LTI_CONSUMER_KEY,
+            client_secret=LTI_CONSUMER_SECRET,
+            signature_type=SIGNATURE_TYPE_BODY,
+        )
+
+    def test_lti_login(self):
+        # The user initiates a login from an external site
+        (uri, _headers, body) = self.lti.sign(
+            uri=LTI_TPA_LOGIN_URL, http_method='POST',
+            headers={'Content-Type': FORM_ENCODED},
+            body={
+                'user_id': LTI_USER_ID,
+                'custom_tpa_next': '/account/finish_auth/?course_id=my_course_id&enrollment_action=enroll',
+            }
+        )
+        login_response = self.client.post(path=uri, content_type=FORM_ENCODED, data=body)
+        # The user should be redirected to the registration form
+        self.assertEqual(login_response.status_code, 302)
+        self.assertTrue(login_response['Location'].endswith(reverse('signin_user')))
+        register_response = self.client.get(login_response['Location'])
+        self.assertEqual(register_response.status_code, 200)
+        self.assertIn('currentProvider&#34;: &#34;LTI Test Tool Consumer&#34;', register_response.content)
```

Comment:
Why is the current provider "LTI Test Tool Consumer" and not one of the other lti providers that you've set up?